### ISSUE TITLE
Can't connect to NodeJS debugger using a docker container

### ISSUE THREAD

**Issue by backops-threed** [Azure/azure-functions-core-tools#2314]
So, my stack uses docker-compose for simplicity changing computers.
The problem is that I can't connect to the debugger for nodejs if it's running inside a docker container.
The worker runs fine but I just can't connect the debugger.

Some important notes:
- I'm using PHPStorm.
- On my container, I'm exposing the 5858 port. It's the same definition that I use for another "normal" (it's not a function) nodejs container, and with that one It's working fine.
- I have 3 different functions, each one with their container and this problem persists inside everyone.
- I have 2 node API's that do not use azure function, running in another two containers and I can connect to their debuggers.
- I can run and attach the debugger locally, on my computer, outside docker, with the same `func -v` as `3.0.2931` and the same `node -v` as `12.18.4` in the docker container
- When I run it on docker, I get the normal log `Debugger listening on ws://127.0.0.1:5858/c30ca721-06c6-4ffe-875f-93f2a22cc882`. Same for local.
- Running locally `funct start`, the PHPStorm Debugger connects fine and logs the following message `LanguageWorkerConsoleLogWorker 7dc4a6c1-1889-43a2-8838-5c85bd093cdf connecting on 127.0.0.1:43697`

So, it seams that `func start` somehow changes the port when attaching the debugger. This port `43697` is random. Every `func start` is a different port, so I don't know how to expose it on the container. The debugger configuration on PHPStorm is set to the port `5858`.

Any ideas? Anyone knows a way to fix that port?

**Comment by anthonychu** [comment]
The random port you are seeing is likely the port that the Azure Functions Node language worker process is using to communicate with the Azure Functions host, and it is not related to the debugging port.

Based on the output, it does look like the debugging port is 5858, and if the same port is exposed to the host machine, I would imagine it should work. However I’m not familiar with PHPStorm. Can you try attaching using one of the other inspector clients listed on this page? https://nodejs.org/en/docs/guides/debugging-getting-started/

**Comment by backops-threed** [comment]
Well, I just found the problem after almost one month living with it and trying to solve time to time.
The Function was listening on `ws://127.0.0.1:5858` and, I just saw, the other docker container that worked, was listening on `ws://0.0.0.0:5858`.
So, changing the environment `languageWorkers:node:arguments` on `local.settings.json` from `--inspect=5858` to `--inspect=0.0.0.0:5858` solved this problem.
Thanks for your time anyway!
